projects/common/src/lib/elements/manage/manage.component.html

```diff
@@ -185,8 +192,30 @@ <h2>Enroll a New Device</h2>
                   placeholder="The Name of your Device"
                   required
                 />
+                
               </mat-form-field>
+              
             </div>
+            <div *ngIf="AddDeviceFormGroup.get('deviceName').invalid && (AddDeviceFormGroup.get('deviceName').dirty || AddDeviceFormGroup.get('deviceName').touched)"
+                      class="alert alert-danger">
+
+                  <mat-error *ngIf="AddDeviceFormGroup.get('deviceName').hasError('required')">
+                    Device name is required
+                  </mat-error>
+
+                  <mat-error *ngIf="AddDeviceFormGroup.get('deviceName').hasError('maxlength')">
+                    Device name cannot be longer than 128 characters
+                  </mat-error>
+
+                  <mat-error *ngIf="AddDeviceFormGroup.get('deviceName').hasError('pattern')">
+                    A case-sensitive string of ASCII 7-bit alphanumeric characters plus certain special characters: - . % _ * ? ! ( ) , : = @ $ '
+                  </mat-error>
+
+                  <mat-error *ngIf="AddDeviceFormGroup.get('deviceName').hasError('duplicateName')">
+                    Device name already exists
+                  </mat-error>

```

Comment:
Same here, instead of these duplicated mat-errors, a single property in the backend would be better:
`public get DeviceNameErrorText(): boolean {
   var errorText: string = null;
   
   if (this.AddDeviceFormGroup.get('deviceName').hasError('required')) {
      errorText = 'Device name is required\r\n';
   }

   if (this.AddDeviceFormGroup.get('deviceName').hasError('maxlength')) {
      errorText = 'Device name cannot be longer than 128 characters\r\n';
   }

...

return errorText;
}`

After that, you can simply do:
`<mat-error>{{ DeviceNameErrorText }}</mat-error>`